src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
 export interface Figure {
-
+  shape: 'triange' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
 export class Triangle {
+  color: string;
+
+  shape: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: string, a: number, b: number, c: number) {
+    this.color = color;
+    this.shape = 'triangle';
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('length is < than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('sides cant form a triangle');
+    }
```

Comment:
would be great to create a separate method for that and just call it inside the constructor